### ISSUE TITLE
Location selector sorting messes up result

### ISSUE THREAD

**Issue by MikeNeilson** [opendcs/opendcs#283]
**Describe the bug**
If one sorts by any column in the location editor a different location is returned

**To Reproduce**
1. Open CompEdit
2. Open Computation
3. select and edit a parameter
4. Sort by base
5. See the base location change in "Add Mask By" but not match the rendered row

**Expected behavior**

Correct element should be presented to downstream uses

**Screenshots**
![image](https://github.com/opendcs/opendcs/assets/7980296/6c715304-75fe-4a69-90b0-ddb26676681e)

**Desktop/ServerSide (please complete the following information):**
 - OS: Windows 10
 - Version: 7.0.8

**Additional context**
This looks to be identical to the items fixed in https://github.com/opendcs/opendcs/pull/265/files, considering these were working I suspect some Java update "corrected" behavior that we either relied on or worked around.

Additionally it appears that JTable can handle sorting on it's own now, we just need to make sure the table model is probably defined and so far as I can tell it is other than getColumnClass not be configured to return a Class appropriate to the actual value.

**Comment by MikeNeilson** [comment]
This also appears to be a good candidate for getting the "Database" to a constructor instead of using Database.getDb() and actually setting up tests without too much difficulty.

While the SiteSelectPanel and Dialog are used in multiple places they would all already have a reference to the Database.

**Comment by hoffmanjs** [comment]
Fixed all (Location, Param, Version, Units, and Time Series Lookup)  of the "Select" button options.  
![image](https://github.com/opendcs/opendcs/assets/122313426/a3259860-f491-41dc-b8d2-f906ddc77089)

When each button is selected it brings up a different dialog box.  All of the dialog boxes had similar sorting issues.  All dialog sorting issues mentioned above were resolved.

**Comment by MikeNeilson** [comment]
@agilmore2 FYI, this likely affects HDB as well, when I was looking through the code a different dialog for location selection is opened. However, it was the dbedit SiteList panel which I believe Scott also fixed so while it needs a quick check it's probably working in the current versions.

**Comment by agilmore2** [comment]
It was messed up when I tried it a week ago from current build, thanks!

**Comment by DanielTOsborne** [comment]
I noticed myself that in the computation list, if you double click a comp, it opens correctly, but if you click "Copy" to duplicate it, the wrong one is copied.
This was with 7.0.11.

**Comment by msweier** [comment]
> I noticed myself that in the computation list, if you double click a comp, it opens correctly, but if you click "Copy" to duplicate it, the wrong one is copied. This was with 7.0.11.

Yes I had the same issue with comps with 7.0.11.. It opens correctly if you double click it, but if you click "Delete" the wrong one was deleted. It was tough to know what one I deleted. I had to wait and see what broke.

**Comment by msweier** [comment]
> Has anyone tried with 7.0.12 yet?

I have not, but I should clarify in 7.0.11 there is a verification prompt that displays the comp it will delete. It just doesn't match the selection after the sort, and if the comps are similar in name it can be easy to click delete. So it should be easy to check if you have 7.0.12 without actually deleting something. Bug should be present in both desktop and server GUIs.
![image](https://github.com/opendcs/opendcs/assets/9432394/c30daa86-7b4f-4b98-8a0a-751cd3417ffc)
![image](https://github.com/opendcs/opendcs/assets/9432394/6e77c478-81b4-4837-a452-b7cbeaa55250)

**Comment by hoffmanjs** [comment]
I have tried this in 7.0.12 desktop version.  It works correctly.
